### PR TITLE
Add Animal metadata to the fcirc advanced search

### PR DESCRIPTION
**Pull request by hepcat72**
## Summary Change Description

I added fields to the fcirc advanced search format and updated the DEVELOPMENT doc to create a section on adding fields.

## Affected Issue Numbers

- Partially addresses #270

## Code Review Notes

Decided to keep the field additions and the show/hide feature separate.

## Checklist

- [x] All issue requirements satisfied (or no linked issues)
- [x] [Linting passes](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting).
- [x] [Migrations created & committed *(or no model changes)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)
- [ ] [Tests implemented *(or no code changes)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
- [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)

### CODE REVIEW

**Comment by hepcat72** [comment]
Looks like I need to update my local superlinter...

**Comment by hepcat72** [comment]
Oh, I just realized that I still need to update the DEVELOPMENT procedure for adding a format to reflect the model "instance" change...  It's only new edits, so I think reviews can still proceed concurrently.

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -71,3 +71,43 @@ Changing a format's content should be indicated by a version number specific to
...
+Note that currently, a composite view class can only contain each model once, or rather, only one link into the same table is supported.
+
+1. `DataRepo/compositeviews.py`
+   - Locate the derived class of BaseSearchView for the output format to which you want to add columns and...
+      - If the model exists in the models datamember
+         - Copy one of the existing fields, paste, and edit.  See field editing notes below.
+      - Else (if the model does not exist in the models datamember)
+         - Copy one of the existing models, paste, and edit.
+         - Always include the primary key as a searchable field.
+         - Add the path to the prefectes array if it is not contained in an existing path in the array already.
```
Comment (jcmatese) [inline DEVELOPMENT.md:86]:
```suggestion
         - Add the path to the prefetches array if it is not contained in an existing path in the array already.
```
Actually not certain how you wanted it phrased, but _seems_ like a typo...

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -71,3 +71,43 @@ Changing a format's content should be indicated by a version number specific to
...
+      - None values should be ensured to display as "None" so they can be differentiated from empty string
+      - Manipulated values (like truncated decimal values) should have a tooltip that shows the full value
+      - Headers should show units in parenthases if a value has units
+
+3. `DataRepo/templates/downloads/<format name>.tsv`
+   - Each template is different, but generally, unless the model doesn't already exist in the template, just add a column to the tab-delimited table.
+   - General guidelines:
+      - Field values should not be manipulated/modified (e.g. do not truncate decimal places) except to match the format supplied by researchers in the loading files (when the stored version in the database differs, e.g. Sample.time_collected or Animal.age, which are saved as time-deltas)
+      - None values should be ensured to display as "None" so they can be differentiated from empty string
+      - Headers should show units in parenthases if a value has units
```
Comment (jcmatese) [inline DEVELOPMENT.md:113]:
```suggestion
      - Headers should show units in parentheses if a value has units
```

**Review by jcmatese — APPROVED** [review]
Seems good to me.  Noted some possible typos in the helpful docs.

**Review by fkang-pu — APPROVED** [review]

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -71,3 +71,43 @@ Changing a format's content should be indicated by a version number specific to
 Be careful that the .tsv file has actual tab characters and note that every newline character in the template will end up in every downloaded file, which is why the lines are so long.
 
 The header line of the .tsv file should be commented.
+
+## How to add/remove columns to an advanced search output format
```
Comment (lparsons) [inline DEVELOPMENT.md:75]: Nice documentation, should be quite useful.

File: DataRepo/compositeviews.py [hunk 4]
```diff
@@ -556,7 +556,7 @@ class PeakGroupsSearchView(BaseSearchView):
                 },
                 "age": {
                     "displayname": "Age (d-hh:mm:ss)",
-                    "searchable": False,
+                    "searchable": False,  # currently no data available for testing and a bug: issue #180
```
Comment (lparsons) [inline DataRepo/compositeviews.py:559]: C: Good point, we should add some test data that include Animal age (perhaps when addressing #180 or #172).

**Review by lparsons — APPROVED** [review]
Especially nice that you took the time to document how to do this. Seems to work well.

File: DEVELOPMENT.md [hunk 5]
```diff
@@ -71,3 +71,43 @@ Changing a format's content should be indicated by a version number specific to
...
+
+Note that currently, a composite view class can only contain each model once, or rather, only one link into the same table is supported.
```
[the remark this answers is not in the document]
Comment (hepcat72) [reply DEVELOPMENT.md:77]: [Yeah](https://github.com/Princeton-LSI-ResearchComputing/tracebase/pull/297#issuecomment-992605924)

**Pull request merged**